portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,19 +14,138 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Task;
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+/** Servlet that returns some example content.*/
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+    int numComments;
+
   @Override
+  // LIST TASKS / COMMENTS
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<Task> tasks = new ArrayList<>();                  
+    
+    for (Entity entity : results.asIterable()) {
+    //for( int i = 0; i < numComments ; i++) {
```

Comment:
Remove commented-out code, here and in a bunch of other spots below. Take a pass at the whole thing to get rid of all the commented-out stuff and also any dead code that's no longer being used. Stuff like this can be annoying, but it's really important to keep code "clean" when you have 10s of engineers submitting 100s of commits to it 😄

Reply:
Also should note - if you have some code that's no longer used, but you wrote it to demonstrate some feature you want to show off / may use it some day, you can leave that in and add a comment explaining why.